main/src/main/scala/sbt/Defaults.scala

```diff
@@ -3666,14 +3491,11 @@ object Classpaths {
       val extResolvers = externalResolvers.value
       val isScala3M123 = ScalaArtifacts.isScala3M123(version)
       val allToolDeps =
-        if (
-          scalaHome.value.isDefined || scalaModuleInfo.value.isEmpty || !managedScalaInstance.value
-        )
-          Nil
-        else if (!isScala3M123 || extResolvers.contains(Resolver.JCenterRepository)) {
```

Comment:
Ok so we won't be able to use scaladoc for `3.0.0-M1`. You probably need to update the Scala version in `project-matrix/jvm-sandwich` scripted test.